### PR TITLE
Raise error when callback has empty arguments in `add_key_event`

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

This PR is to save us in #4750. We were too lazy to read the documentation and it took us quite some time to solve the problem.

### Details

- None

### CODE REVIEW

File: pyvista/plotting/render_window_interactor.py [hunk 1]
```diff
@@ -102,6 +103,8 @@ def add_key_event(self, key, callback):
         """
         if not callable(callback):
             raise TypeError('callback must be callable.')
+        if len(signature(callback).parameters) > 0:
```
Comment (dcbr) [inline pyvista/plotting/render_window_interactor.py:106]: Isn't this a bit too restrictive? If the callback provides default values for all arguments, it shouldn't be a problem right? (But I'm not sure how you would check that)

Comment (tkoyama010) [reply pyvista/plotting/render_window_interactor.py:106]: Thank you for your comment. That is a very good point. I had completely missed it's use case. The following code will accomplish that.
```suggestion
        for param in signature(callback).parameters.values():
            if (param.default is param.empty):
```

**Review by dcbr — APPROVED** [review]
LGTM, only a small suggestion for the docs; feel free to ignore.

File: pyvista/plotting/render_window_interactor.py [hunk 2]
```diff
@@ -97,11 +98,14 @@ def add_key_event(self, key, callback):
...
         callback : callable
-            A callable that takes no arguments.
+            A callable that takes no empty arguments.
 
         """
         if not callable(callback):
             raise TypeError('callback must be callable.')
+        for param in signature(callback).parameters.values():
+            if param.default is param.empty:
+                raise TypeError('callback must not have any empty arguments.')
```
Comment (akaszynski) [inline pyvista/plotting/render_window_interactor.py:108]:
```suggestion
                TypeError(f'`callback` must not have any arguments without default values.')
```

**Review by akaszynski — APPROVED** [review]

**Review by dcbr — APPROVED** [review]

**Pull request merged**